cmd/sqlcmd/sqlcmd.go

```diff
@@ -117,25 +118,140 @@ func (a SQLCmdArguments) authenticationMethod(hasPassword bool) string {
+	var variables map[string]string
+	rootCmd.Flags().StringToStringVarP(&args.Variables, "Variables", "v", variables, "Creates a sqlcmd scripting variable that can be used in a sqlcmd script. Enclose the value in quotation marks if the value contains spaces. You can specify multiple var=values values. If there are errors in any of the values specified, sqlcmd generates an error message and then exits")
+	rootCmd.Flags().IntVarP(&args.PacketSize, "PacketSize", "a", 0, "Requests a packet of a different size. This option sets the sqlcmd scripting variable SQLCMDPACKETSIZE. packet_size must be a value between 512 and 32767. The default = 4096. A larger packet size can enhance performance for execution of scripts that have lots of SQL statements between GO commands. You can request a larger packet size. However, if the request is denied, sqlcmd uses the server default for packet size.")
+	rootCmd.Flags().IntVarP(&args.LoginTimeout, "LoginTimeOut", "l", -1, "Specifies the number of seconds before a sqlcmd login to the go-mssqldb driver times out when you try to connect to a server. This option sets the sqlcmd scripting variable SQLCMDLOGINTIMEOUT. The default value is 30. 0 means infinite.")
+	rootCmd.Flags().StringVarP(&args.WorkstationName, "WorkstationName", "H", "", "This option sets the sqlcmd scripting variable SQLCMDWORKSTATION. The workstation name is listed in the hostname column of the sys.sysprocesses catalog view and can be returned using the stored procedure sp_who. If this option is not specified, the default is the current computer name. This name can be used to identify different sqlcmd sessions.")
+
+	rootCmd.Flags().StringVarP(&args.ApplicationIntent, "ApplicationIntent", "K", "default", "Declares the application workload type when connecting to a server. The only currently supported value is ReadOnly. If -K is not specified, the sqlcmd utility will not support connectivity to a secondary replica in an Always On availability group")
+	rootCmd.Flags().StringVarP(&args.EncryptConnection, "EncryptConnection", "N", "default", "This switch is used by the client to request an encrypted connection.")
+	rootCmd.Flags().StringVarP(&args.Format, "Format", "F", "horiz", "Specifies the formatting for results.")
+	rootCmd.Flags().IntVarP(&args.ErrorsToStderr, "ErrorsToStderr", "r", -1, "Controls which error messages are sent to stdout. Messages that have severity level greater than or equal to this level are sent.")
+
+	//Adding a validator for checking the enum flags
+	rootCmd.Flags().SetNormalizeFunc(func(f *pflag.FlagSet, name string) pflag.NormalizedName {
+		switch name {
+		case "ApplicationIntent":
+			value := getFlagValueByName(f, name)
+			switch value {
+			case "default", "ReadOnly":
+				return pflag.NormalizedName(name)
+			default:
+				fmt.Fprintf(os.Stderr, "Invalid value for %s flag. Allowed values are default and ReadOnly.\n", name)
```

Comment:
>Invalid value for %s flag. Allowed values are default and ReadOnly.\n", name) [](http://example.com/codeflow?start=28&length=77)

Use the same error messages that the ODBC sqlcmd uses

eg

Sqlcmd: Option -K: Unexpected argument: foobar